compiler/rustc_passes/src/dead.rs

```diff
@@ -647,6 +650,29 @@ impl<'tcx> Visitor<'tcx> for MarkSymbolVisitor<'tcx> {
 
         self.in_pat = in_pat;
     }
+
+    fn visit_poly_trait_ref(&mut self, t: &'tcx hir::PolyTraitRef<'tcx>) {
+        // Mark the assoc const appears in poly-trait-ref live
+        if let Some(pathsegment) = t.trait_ref.path.segments.last()
```

Comment:
In theory, constraints can appear in non-last segments, and outside of poly trait refs too.
So, a safer choice here would probably be using `visit_assoc_item_constraint`.